tyrian/js/src/main/scala/tyrian/cmds/FileReader.scala

```diff
@@ -53,16 +54,36 @@ object FileReader:
         try Result.File(n, p, d.asInstanceOf[String])
         catch case _ => Result.Error("File is not text")
 
-    readFile(file, true)(cast andThen resultToMessage)
+    readFile(file, ReadType.AsText)(cast andThen resultToMessage)
 
-  private def readFromInputField[F[_]: Async, Msg](fileInputFieldId: String, isText: Boolean)(
+  /** Reads an input file from an input field as bytes */
+  def readBytes[F[_]: Async, Msg](inputFieldId: String)(resultToMessage: Result[Vector[Byte]] => Msg): Cmd[F, Msg] =
+    val cast: Result[js.Any] => Result[Vector[Byte]] =
+      case Result.Error(msg) => Result.Error(msg)
+      case Result.File(n, p, d) =>
+        try Result.File(n, p, new typedarray.Int8Array(d.asInstanceOf[typedarray.ArrayBuffer]).toVector)
+        catch case _ => Result.Error("File is not bytes")
```

Comment:
Not the greatest message here, all files are bytes... but I'm not sure what else would be appropriate?

Reply:
I would go with something like "Could not cast loaded file data to byte array". Specific about what went wrong but vague on the implementation details.